databricks/koalas/frame.py

```diff
@@ -4908,6 +4909,155 @@ def drop_duplicates(self, subset=None, inplace=False):
+
+        >>> df.rank(method='min').sort_index()
+             A    B
+        0  1.0  4.0
+        1  2.0  3.0
+        2  2.0  2.0
+        3  4.0  1.0
+
+        >>> df.rank(method='dense').sort_index()
+             A    B
+        0  1.0  4.0
+        1  2.0  3.0
+        2  2.0  2.0
+        3  3.0  1.0
+
+        >>> df = ks.DataFrame({'A': [1, 2, None, 3], 'B': [4, None, 2, 1]}, columns= ['A', 'B'])
+        >>> df
+             A    B
+        0  1.0  4.0
+        1  2.0  NaN
+        2  NaN  2.0
+        3  3.0  1.0
+
+        >>> df.rank().sort_index()
+             A    B
+        0  1.0  3.0
+        1  2.0  NaN
+        2  NaN  2.0
+        3  3.0  1.0
+
+        >>> df.rank(na_option='bottom').sort_index()
+             A    B
+        0  1.0  3.0
+        1  2.0  4.0
+        2  4.0  2.0
+        3  3.0  1.0
+        """
+        if method not in ['average', 'min', 'max', 'first', 'dense']:
+            msg = "method must be one of 'average', 'min', 'max', 'first', 'dense'"
+            raise ValueError(msg)
+
+        if na_option not in ['keep', 'top', 'bottom']:
+            msg = "na_option must be one of 'keep', 'top', 'bottom'"
+            raise ValueError(msg)
+
+        if ascending:
+            asc_func = spark.functions.asc
+        else:
+            asc_func = spark.functions.desc
+
+        index_column = self._internal.index_columns[0]
+        data_columns = self._internal.data_columns
+        sdf = self._sdf
+
+        for column_name in data_columns:
+            if na_option == 'top':
+                if ascending:
+                    fill_value = sdf.agg({column_name: 'min'}).collect()[0][0] - 1
+                else:
+                    fill_value = sdf.agg({column_name: 'max'}).collect()[0][0] + 1
+                sdf = sdf.fillna({column_name: fill_value})
+            elif na_option == 'bottom':
+                if ascending:
+                    fill_value = sdf.agg({column_name: 'max'}).collect()[0][0] + 1
+                else:
+                    fill_value = sdf.agg({column_name: 'min'}).collect()[0][0] - 1
+                sdf = sdf.fillna({column_name: fill_value})
+            else:
+                null_number = sdf.filter(F.col(column_name).isNull()).count()
+                if null_number > 0:
+                    sdf = sdf.withColumn(column_name + '_tmp', F.col(column_name))
+
+            if method == 'first':
+                window = Window.orderBy(asc_func(column_name), asc_func(index_column))\
+                    .rowsBetween(Window.unboundedPreceding, Window.currentRow)
+                sdf = sdf.withColumn(column_name, F.row_number().over(window))
+            elif method == 'dense':
+                window = Window.orderBy(asc_func(column_name))\
+                    .rowsBetween(Window.unboundedPreceding, Window.currentRow)
+                sdf = sdf.withColumn(column_name, F.dense_rank().over(window))
+            else:
+                if method == 'average':
```

Comment:
Hey, I meant I was wondering if we can deduplicate this code paths:

https://github.com/databricks/koalas/blob/1dbf5dea29a0bb171ddad4aff802b57e6658cafe/databricks/koalas/series.py#L2239-L2252

So that Series and DataFrame both have the same `rank` functionalities.